---
title: "SDH inhibition snMultiome RNA & ATAC-seq"
author: "Dakota Nuttall"
date: "2025-06-17"
R-version: 4.3.3
---
###Cardiomyocyte Analysis###

```{r echo=TRUE}
set.seed(1234)
library(Seurat)
library(Signac)
library(chromVAR)
library(dplyr)
library(reticulate)
library(hdf5r)
library(scales)
library(ggplot2)
library(openxlsx)
library(readxl)
library(EnsDb.Mmusculus.v79)
library(BSgenome.Mmusculus.UCSC.mm10)
library(org.Mm.eg.db)
library(JASPAR2020)
library(TFBSTools)
library(harmony)
library(pheatmap)
library(EnhancedVolcano)
library(ggrepel)
library(tidyr)
library(gridExtra)
library(ggpubr)
```
####  Load data  ####

```{r echo=TRUE}
#Load integrated file
aggregated_samples <- readRDS("aggregated_wnn.rds")

#Subset Cardiomyocyte (CM) population
aggregated_Cardiomyocytes <- subset(aggregated_samples, idents = "Cardiomyocytes")
aggregated_Cardiomyocytes <- SCTransform(aggregated_Cardiomyocytes, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)

DefaultAssay(aggregated_Cardiomyocytes) <- "SCT"


```
##

```{r echo=TRUE}
#Establish new RNA, ATAC and WNN umaps
ElbowPlot(aggregated_Cardiomyocytes, reduction = "pca", ndims = 50)
aggregated_Cardiomyocytes <- RunUMAP(aggregated_Cardiomyocytes, dims = 1:50, reduction = "harmony_pca",reduction.name = "umap_RNA", reduction.key = "UMAP")

aggregated_Cardiomyocytes <- PrepSCTFindMarkers(aggregated_Cardiomyocytes, assay = "SCT",verbose = T)

DefaultAssay(aggregated_Cardiomyocytes) <- "peakunion"

aggregated_Cardiomyocytes <- FindTopFeatures(aggregated_Cardiomyocytes, min.cutoff = 10)
aggregated_Cardiomyocytes<- RunTFIDF(aggregated_Cardiomyocytes,)
aggregated_Cardiomyocytes <- RunSVD(aggregated_Cardiomyocytes)

ElbowPlot(aggregated_Cardiomyocytes, ndims = 50, reduction = "harmony_lsi")

aggregated_Cardiomyocytes <- RunUMAP(object = aggregated_Cardiomyocytes, reduction = 'harmony_lsi', dims = 2:50, reduction.name = "umap_peakunion", reduction.key = "peakunionUMAP_")


DefaultAssay(aggregated_Cardiomyocytes) <- "SCT"
FindDimNumber(aggregated_Cardiomyocytes,reduction = "lsi")
aggregated_Cardiomyocytes <- FindMultiModalNeighbors(
  object = aggregated_Cardiomyocytes,
  reduction.list = list("harmony_pca", "lsi"), 
    dims.list = list(1:50, 2:50),
  modality.weight.name = c("SCT.weight","peakunion.weight"),
  verbose = TRUE
)
aggregated_Cardiomyocytes <- RunUMAP(
  object = aggregated_Cardiomyocytes,
  nn.name = "weighted.nn",
  assay = "SCT",
  reduction.name = "WNN_umap",
  reduction.key = "WNN_UMAP_",
  verbose = TRUE,
)

aggregated_Cardiomyocytes<- FindClusters(aggregated_Cardiomyocytes, graph.name = "wsnn", algorithm = 3, resolution = seq(.1,.5,.1))

quick_CT_plot(aggregated_Cardiomyocytes, 2,"WNN_umap",legend = F,label = T, font.size = 8, resolution = "SCT_snn_res.0.2")
quick_cond_plot(aggregated_Cardiomyocytes, 2,"WNN_umap",legend = F)

```
##
Find differentially expressed genes (DEGs) and identitify ambiguous clusters
```{r echo=TRUE}
#Identify upregulated genes
aggregated.Cardiomyocytesmarkers <- FindAllMarkers(aggregated_Cardiomyocytes, only.pos = F, min.pct = 0.1, logfc.threshold = .3, test.use = "MAST", recorrect_umi = TRUE, assay = "SCT")
aggregated.Cardiomyocytesmarkers <- subset(aggregated.Cardiomyocytesmarkers, p_val_adj < 0.05)

#Identify abnormal RNA counts
VlnPlot(aggregated_Cardiomyocytes, "nCount_RNA",pt.size = .1, cols = clust_colors) + NoLegend()+ xlab(NULL) + theme(text=element_text(size=14),axis.text=element_text(size=18,face="bold"),axis.title=element_text(size=20,face="bold"))


```
##Cluster '1' demonstrated abnormally low RNA UMI counts for CMs as well as expression of Cardiac Fibroblast markers like Pdgfra along with CM markers so we will relable that cluster as 'Ambiguous' subset/remove it from the CM dataset 

```{r echo=TRUE}
aggregated_Cardiomyocytes <- subset(aggregated_Cardiomyocytes, idents = "1", invert = T)

#repreform normalization and UMAPs
aggregated_Cardiomyocytes <- SCTransform(aggregated_Cardiomyocytes, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)

DefaultAssay(aggregated_Cardiomyocytes) <- "SCT"

ElbowPlot(aggregated_Cardiomyocytes, reduction = "pca", ndims = 50)
aggregated_Cardiomyocytes <- RunUMAP(aggregated_Cardiomyocytes, dims = 1:50, reduction = "harmony_pca",reduction.name = "umap_RNA", reduction.key = "UMAP")

aggregated_Cardiomyocytes <- PrepSCTFindMarkers(aggregated_Cardiomyocytes, assay = "SCT",verbose = T)

DefaultAssay(aggregated_Cardiomyocytes) <- "peakunion"

aggregated_Cardiomyocytes <- FindTopFeatures(aggregated_Cardiomyocytes, min.cutoff = 10)
aggregated_Cardiomyocytes<- RunTFIDF(aggregated_Cardiomyocytes,)
aggregated_Cardiomyocytes <- RunSVD(aggregated_Cardiomyocytes)

ElbowPlot(aggregated_Cardiomyocytes, ndims = 50, reduction = "harmony_lsi")

aggregated_Cardiomyocytes <- RunUMAP(object = aggregated_Cardiomyocytes, reduction = 'harmony_lsi', dims = 2:50, reduction.name = "umap_peakunion", reduction.key = "peakunionUMAP_")


DefaultAssay(aggregated_Cardiomyocytes) <- "SCT"
FindDimNumber(aggregated_Cardiomyocytes,reduction = "lsi")
aggregated_Cardiomyocytes <- FindMultiModalNeighbors(
  object = aggregated_Cardiomyocytes,
  reduction.list = list("harmony_pca", "lsi"), 
    dims.list = list(1:50, 2:50),
  modality.weight.name = c("SCT.weight","peakunion.weight"),
  verbose = TRUE
)
aggregated_Cardiomyocytes <- RunUMAP(
  object = aggregated_Cardiomyocytes,
  nn.name = "weighted.nn",
  assay = "SCT",
  reduction.name = "WNN_umap",
  reduction.key = "WNN_UMAP_",
  verbose = TRUE,
)

aggregated_Cardiomyocytes<- FindClusters(aggregated_Cardiomyocytes, graph.name = "wsnn", algorithm = 3, resolution = seq(.1,.5,.1))

Cardiomyocytes_cell_types <- c("CM1","CM3","CM2") 
aggregated_Cardiomyocytes@active.ident<-factor(aggregated_Cardiomyocytes@active.ident, labels = Cardiomyocytes_cell_types)

Cardiomyocytes_cell_levels <- c("CM1","CM2","CM3")
aggregated_Cardiomyocytes@active.ident<-factor(aggregated_Cardiomyocytes@active.ident, levels =  Cardiomyocytes_cell_levels)

clust_colors <- c("darkorange2","mediumorchid2","steelblue3","olivedrab","indianred1","hotpink","darkred","green3","grey")
treatment_colors <-c("lightskyblue","pink", "steelblue4","red3")                                         
quick_CT_plot(aggregated_Cardiomyocytes, 2,"WNN_umap",legend = F,label = T, font.size = 8)
ggsave("Cardiomyocytes_umap_sub.tiff", height = 5,width = 5)
quick_cond_plot(aggregated_Cardiomyocytes, 2,"WNN_umap",legend = F)
ggsave("Cardiomyocytes_umap_cond.tiff", height = 5,width = 5)


```
## Find DEGs once again for Volcano plots and Gene Ontology (see GO script)

```{r echo=TRUE}
aggregated.Cardiomyocytesmarkers <- FindAllMarkers(aggregated_Cardiomyocytes, only.pos = F, min.pct = 0.1, logfc.threshold = .3, test.use = "MAST", recorrect_umi = TRUE, assay = "SCT")
aggregated.Cardiomyocytesmarkers <- subset(aggregated.Cardiomyocytesmarkers, p_val_adj < 0.05)
write.xlsx(aggregated.Cardiomyocytesmarkers,"aggregated_Cardiomyocytes_DEGs.xlsx")
```
## Volcano plot analysis

```{r echo=TRUE}

##For CM2
DefaultAssay(aggregated_Cardiomyocytes) <- "SCT"
CM2_marker <- FindMarkers(aggregated_Cardiomyocytes, ident.1 = "CM2", only.pos = F, min.pct = 0.1, logfc.threshold = .3, test.use = "MAST", recorrect_umi = TRUE, assay = "SCT" )
CM2_marker$gene <- rownames(CM2_marker)
CM2_marker <- CM2_marker[!grepl("mt-",CM2_marker$gene),]
CM2_marker_sig <- subset(CM2_marker, p_val_adj < 0.05)

lab_italics <- paste0("italic('", rownames(CM2_marker), "')")
select_labels <-c(head(row.names(CM2_marker),n=10), 'Myh7','Tgfb2','Ccn2',head(row.names(subset(CM2_marker, avg_log2FC < 0)),n=10))
select_labels <- paste0("italic('",c(head(row.names(CM2_marker),n=15), 'Myh7','Tgfb2','Ccn2',head(row.names(subset(CM2_marker, avg_log2FC < 0)),n=10)), "')")

#find number of significant upregulated and downregulated genes
sum(CM2_marker_sig$avg_log2FC < 0)
sum(CM2_marker_sig$avg_log2FC > 0)

up_down_volcano(CM2_marker, selectLab =  select_labels,italicLab = lab_italics,up_label = "CM2 Upregulated Genes (98 significant)",down_label = "CM2 Downregulated Genes (61 significant)",volcano_name = "CM2_volcano.tiff", up_x = 6,down_x = -4.5 )

##For CM3
DefaultAssay(aggregated_Cardiomyocytes) <- "SCT"
CM3_marker <- FindMarkers(aggregated_Cardiomyocytes, ident.1 = "CM3", only.pos = F, min.pct = 0.1, logfc.threshold = .3, test.use = "MAST", recorrect_umi = TRUE, assay = "SCT" )
CM3_marker$gene <- rownames(CM3_marker)
CM3_marker <- CM3_marker[!grepl("mt-",CM3_marker$gene),]
CM3_marker_sig <- subset(CM3_marker, p_val_adj < 0.05)

lab_italics <- paste0("italic('", rownames(CM3_marker), "')")
select_labels <-c(head(row.names(CM3_marker),n=10), 'Sesn1','Tcap','Ccn1',head(row.names(subset(CM3_marker, avg_log2FC < 0)),n=10))
select_labels <- paste0("italic('",c(head(row.names(CM3_marker),n=15), 'Sesn1','Tcap','Ccn1',head(row.names(subset(CM3_marker, avg_log2FC < 0)),n=10)), "')")

#find number of significant upregulated and downregulated genes
sum(CM3_marker_sig$avg_log2FC < 0)
sum(CM3_marker_sig$avg_log2FC > 0)

up_down_volcano(CM3_marker, selectLab =  select_labels,italicLab = lab_italics,up_label = "CM3 Upregulated Genes (194 significant)",down_label = "CM3 Downregulated Genes (34 significant)",volcano_name = "CM3_volcano.tiff", up_x = 6,down_x = -4.5 )
```
## Identify upregulated ATAC peaks

```{r echo=TRUE}
annotation <- GetGRangesFromEnsDb(ensdb = EnsDb.Mmusculus.v79)
annotation <- renameSeqlevels(annotation, mapSeqlevels(seqlevels(annotation), "UCSC"))
DefaultAssay(aggregated_Cardiomyocytes) <- 'peakunion'
Annotation(aggregated_Cardiomyocytes) <- annotation


# Linear regression is commonly used to analyze ATAC Peaks
da_unionpeak <- FindMarkers(aggregated_Cardiomyocytes,
  ident.1 = "CM2",
  test.use = 'LR',
  min.pct = 0.1, only.pos = F,
  logfc.threshold = 0.3, 
  latent.vars = 'nCount_peakunion',
)
da_unionpeak <- da_unionpeak[da_unionpeak$p_val_adj < 0.05 , ]
da_unionpeak$region <- rownames(da_unionpeak)
closest_genes <- ClosestFeature(aggregated_Cardiomyocytes, regions = rownames(da_unionpeak))
da_unionpeak <- do.call(cbind, list(da_unionpeak,closest_genes)) 
head(da_unionpeak)
write.xlsx(da_unionpeak,"CM2_unionpeak.xlsx")

open_CM2 <- da_unionpeak$region[da_unionpeak$avg_log2FC > 0.3 ]
open_CM3 <- da_unionpeak$region[da_unionpeak$avg_log2FC > 0.3 ]

#This function identifies genes that are closest to the peak
closest_genes_CM2 <- ClosestFeature(aggregated_Cardiomyocytes, regions = open_CM2)
closest_genes_CM3 <- ClosestFeature(aggregated_Cardiomyocytes, regions = open_CM3)

up_genes_CM2 <- CM2_marker_sub$gene[CM2_marker_sub$avg_log2FC > 0.3 ]
up_genes_CM3 <- CM3_marker_sub$gene[CM3_marker_sub$avg_log2FC > 0.3 ]

#Identify genes that were significantly upregulated that also have a nearby significantly accessible peak
up_RNA_ATAC_CM2 <- intersect(up_genes_CM2,closest_genes_CM2$gene_name)
up_RNA_ATAC_CM2

up_RNA_ATAC_CM3 <- intersect(up_genes_CM3,closest_genes_CM3$gene_name)
up_RNA_ATAC_CM3


#visualize differentially accessible peaks that have nearby significantly upregulated gene

regions_highlight <- subsetByOverlaps(StringToGRanges(open_CM3), LookupGeneCoords(aggregated_Cardiomyocytes, "Pdk4"),maxgap = 1000)
# Can use this for manual selection: regions_highlight <- StringToGRanges("region")
regions_highlight$color <- "skyblue4"

p<-CoveragePlot(
  aggregated_Cardiomyocytes,
  region = "Pdk4",
  region.highlight = regions_highlight,
  extend.upstream = 2000,
  extend.downstream = 2000,  
)

p & scale_fill_manual(values =  clust_colors) & theme( strip.text =element_text(size=16, face = "bold"),axis.title=element_text(size=14,face="bold"),legend.text  = element_text(face="bold", size = 12))
ggsave("CM_Pdk4_peak.tiff",height = 5,width = 6)

## repeat for Ccnd3

```
## Motif analysis
```{r echo=TRUE}
#load motifs 
pfm <- getMatrixSet(
  x = JASPAR2020,
  opts = list(collection = "CORE", tax_group = 'vertebrates', all_versions = FALSE)
)

# first compute the GC content for each peak
aggregated_Cardiomyocytes <- RegionStats(aggregated_Cardiomyocytes, genome = BSgenome.Mmusculus.UCSC.mm10)

# add motifs to CM dataset

aggregated_Cardiomyocytes <- AddMotifs(
  object = aggregated_Cardiomyocytes,
  genome = BSgenome.Mmusculus.UCSC.mm10,
  pfm = pfm,
)

##Analyze motif activity
aggregated_Cardiomyocytes <- RunChromVAR(
  object = aggregated_Cardiomyocytes,
  genome = BSgenome.Mmusculus.UCSC.mm10,
  assay = "peakunion"
)

DefaultAssay(aggregated_Cardiomyocytes) <- "chromvar"

da_peakunion_CM <- FindAllMarkers(
  object = aggregated_Cardiomyocytes,
  only.pos = TRUE,
  mean.fxn = rowMeans,
  fc.name = "avg_diff"
)

da_peakunion_CM$motif.names <-  ConvertMotifID(aggregated_Cardiomyocytes,assay = "peakunion",id =row.names(da_peakunion_CM))

da_peakunion_CM<- na.omit(da_peakunion_CM)
da_peakunion_CM <- subset(da_peakunion_CM, p_val_adj < 0.05)
write.xlsx(da_peakunion_CM, "Cardiomyocytes_motifs.xlsx")

MotifPlot(
  object = aggregated_Cardiomyocytes,
  motifs = row.names(da_peakunion_CM[c('MA0660.1','MA1103.2','MA0773.1','MA0497.1','MA0052.4','MA0852.2'),]), assay = "peakunion"
)

##Visualize individual motifs with violin plots and umaps

v <- VlnPlot(aggregated_Cardiomyocytes, "MA0852.2", pt.size = 0.1, assay = "chromvar", cols = clust_colors) + NoLegend()+ xlab(NULL) + theme(text=element_text(size=20),axis.text=element_text(size=18,face="bold"),axis.title=element_text(size=18,face="bold"))+ labs(title = "Foxk1 Motif")
v

ggsave(filename = "CM_Foxk1_Motif_vln.tiff",plot = v, height = 5, width = 5)

p <- FeaturePlot(
  object = aggregated_Cardiomyocytes,
  features = "MA0852.2",
  pt.size = 2,
  reduction = "WNN_umap",
  min.cutoff = 0, cols = c("lightgrey", "darkred")
) + labs(title = "Foxk1 Motif")+
  theme(text=element_text(size=24),legend.text  = element_text(face="bold", size = 16), axis.text=element_blank(),axis.ticks=element_blank(), axis.line = element_line(linewidth =  1))+ xlab(NULL) + ylab(NULL) +theme(axis.text=element_blank(),axis.title=element_text(size=20,face="bold"),legend.text  = element_text(face="bold", size = 16))
p
ggsave(filename = "CM_Foxk1_Motif_umap.tiff",plot = p, height = 6, width = 6)
##Repeat for the Mef2 motifs and Foxk2


```
## Proportion plots and subcluster heatmaps
```{r echo=TRUE}

aggregated_Cardiomyocytes$Cell_state <- Idents(aggregated_Cardiomyocytes)
#These functions can be found in the helper functions script
proportion_plots(aggregated_Cardiomyocytes, cond_color = treatment_colors, clust_color = clust_colors, name.clust = "CM_prop_sub.tiff", name.cond = "CM_prop_cond.tiff",height = 4, width = 4)

prop_heatmap(aggregated_Cardiomyocytes, "CM_heat_prop.tiff")

#Save dataset
save(aggregated_Cardiomyocytes,"Cardiomyocytes.rds")
```

