---
title: "SDH inhibition snMultiome RNA & ATAC-seq"
author: "Dakota Nuttall"
date: "2025-06-17"
R-version: 4.3.3
---

###Global Analysis###
```{r echo=TRUE}
set.seed(1234)
library(Seurat)
library(SeuratObject)
library(Signac)
library(harmony)
library(dplyr)
library(reticulate)
library(hdf5r)
library(scales)
library(ggplot2)
library(EnhancedVolcano)
library(openxlsx)
library(readxl)
library(GenomeInfoDb)
library(EnsDb.Mmusculus.v79)
library(BSgenome.Mmusculus.UCSC.mm10)
library(org.Mm.eg.db)
library(JASPAR2024)
library(TFBSTools)
library(DropletUtils)
library(Gviz)
library(ggpubr)
```
####  Load data  ####

```{r echo=TRUE}
#Load integrated file
aggregated_samples <- readRDS("aggregated_samples.rds")

#Apply subcluster identites to Cell state metadata
CM<- readRDS("Cardiomyocytes.rds")
CF<- readRDS("Cardiac_Fibroblasts.rds")
Myeloid <- readRDS("Myeloid.rds")
Mural <- readRDS("Mural.rds")
EC <- readRDS("Endothelial_Cells.rds")
Lymph <- readRDS("aggregated_Lymph_union.rds")

aggregated_samples$Cell_state <- "Ambiguous"
aggregated_samples$Cell_state[Cells(CM)] <- paste(Idents(CM))
aggregated_samples$Cell_state[Cells(CF)] <- paste(Idents(CF))
aggregated_samples$Cell_state[Cells(Myeloid)] <- paste(Idents(Myeloid))
aggregated_samples$Cell_state[Cells(Mural)] <- paste(Idents(Mural))
aggregated_samples$Cell_state[Cells(EC)] <- paste(Idents(EC))
aggregated_samples$Cell_state[Cells(subset(Lymph))] <- paste(Idents(Lymph))
aggregated_samples$Cell_state[Cells(subset(aggregated_samples, Cell_type == "Epicardial Cells"))] <- "EpiC"
aggregated_samples$Cell_state[Cells(subset(aggregated_samples, Cell_type == "Schwann Cells"))] <- "Schwann"

# Apply 'Ambiguous' cell state to Cell type as well
aggregated_samples$Cell_type[Cells(subset(aggregated_samples, Cell_state == "Ambiguous"))] <- "Ambiguous"

##Visualize
clust_colors <- c("darkorange2","mediumorchid2","steelblue3","olivedrab","indianred1","hotpink","darkred","green3","grey")
treatment_colors <-c("lightskyblue","pink", "steelblue4","red3")

#Functions below found in helper functions script
quick_CT_plot(object = aggregated_samples,pt.size = .1,reduction = "umap_ATAC", legend = F, label = F, font.size = 6)
ggsave("Global_ATAC_umap_sub.tiff", height = 7,width = 7)
quick_cond_plot(object = aggregated_samples,pt.size = .1,reduction = "WNN_umap", legend = F)
ggsave("Global_umap_cond.tiff", height = 7,width = 7)

#Save final complete version of aggregated data
saveRDS(aggregated_samples,"aggregated_final.rds")

```
##Prepare database for visualizing cluster/annotation efficiency and accuracy

```{r echo=TRUE}
# Remove 'Ambiguous' and "Schwann Cell' cluster (Schwann population is too small for proper analysis)
DefaultAssay(aggregated_samples) <- "SCT"
aggregated_samples <-subset(aggregated_samples,idents = c("Ambiguous","Schwann Cells"), invert = T)
Aggregated_cell_levels <- c("Cardiomyocytes","Cardiac Fibroblasts","Endothelial Cells","Mural Cells","Myeloid Cells","Epicardial Cells","Lymphoid Cells")
aggregated_samples@active.ident<-factor(aggregated_samples@active.ident, levels = Aggregated_cell_levels)

aggregated_samples <- SCTransform(aggregated_samples, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)
aggregated_samples <- PrepSCTFindMarkers(aggregated_samples, assay = "SCT",verbose = T)

aggregated.markers <- FindAllMarkers(aggregated_samples, only.pos = TRUE, logfc.threshold = 1, return.thresh = 0.1, min.pct = 0.5, test.use = "MAST", recorrect_umi = FALSE)

aggregated.markers <- subset(aggregated.markers, p_val_adj < 0.05)

aggregated.markers %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > 1) %>%
  slice_head(n = 20) %>%
  ungroup() -> top20

#Use the Random_IDs function from the helper functions to randomly select 100 nuclei from each cluster for heat map plotting
agg_random_IDS <-subset(aggregated_samples, cells = Random_IDs(aggregated_samples))
agg_random_IDS <- SCTransform(agg_random_IDS, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)
agg_random_IDS <- PrepSCTFindMarkers(agg_random_IDS, assay = "SCT",verbose = T)
table(agg_random_IDS@active.ident)
agg_random_IDS@active.ident <- factor(agg_random_IDS@active.ident, labels  = c("CM","CF","EC","Mural","Mye","Epi","Lym"))

```
## Confirm Cell type clusters have distinct RNA signatures
```{r echo=TRUE}
aggregated.markers <- FindAllMarkers(agg_random_IDS, only.pos = TRUE, logfc.threshold = 1, return.thresh = 0.1, min.pct = 0.5, test.use = "MAST", recorrect_umi = FALSE)

aggregated.markers <- subset(aggregated.markers, p_val_adj < 0.05)
write.xlsx(aggregated.markers,"global_RNA_markers.xlsx")

aggregated.markers %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > 1) %>%
  slice_head(n = 20) %>%
  ungroup() -> top20

#Visualize Signature Gene expression profile for each cell type 
DoHeatmap(agg_random_IDS, group.colors = clust_colors,features = top20$gene, label = F, slot = "data",disp.max = 4)  + scale_y_discrete(labels = NULL) + scale_fill_gradient(low = "white", high = "black") + theme(legend.text =element_text(size=12, face = "bold"))
ggsave("Gene_cluster_heatmap.tiff", height = 5,width = 5)

#Visualize Cell type specific gene
v1<-
  VlnPlot(object = aggregated_samples, features = "Myh6", pt.size = .01, cols = clust_colors,)+ NoLegend() + coord_flip() + scale_x_discrete(limits=rev) + theme(axis.text.y  = element_blank(), axis.title.y = element_blank(),  )
##Repeat for each of the following:
#v1-Myh6 (CM),v2-Pdgfra(CF),v3-Cdh5(EC),v4-Pdgfrb(Mye),v5-Cd14(Mural),v6-Upk3b(EpiC),v7-Ms4a1(Lymph)

#Merge into one figure
combined_v <- ggarrange(v1,v2,v3,v4,v5,v6,v7, ncol = 7)
ggsave("global_GEX.tiff", combined_v, width = 20, height = 5)

```
##Confirm Cell type clusters have distinct ATAC signatures
```{r echo=TRUE}
annotation <- GetGRangesFromEnsDb(ensdb = EnsDb.Mmusculus.v79)
annotation <- renameSeqlevels(annotation, mapSeqlevels(seqlevels(annotation), "UCSC"))

DefaultAssay(aggregated_wnn) <- "peakunion"
Annotation(aggregated_wnn) <- annotation

DefaultAssay(agg_random_IDS) <- "peakunion"

agg_random_IDS <- FindTopFeatures(agg_random_IDS, min.cutoff = 10)
agg_random_IDS<- RunTFIDF(agg_random_IDS,)
agg_random_IDS <- RunSVD(agg_random_IDS)

agg.peak.markers <- FindAllMarkers(agg_random_IDS,only.pos = T, min.pct = 0.1, logfc.threshold = .3, test.use = "LR", recorrect_umi = TRUE, assay = "peakunion")
epi.markers<- FindMarkers(agg_random_IDS,ident.1 = "Epi",only.pos = T, min.pct = 0.1, logfc.threshold = .3, test.use = "LR", recorrect_umi = TRUE, assay = "peakunion")
agg.peak.markers <- subset(agg.peak.markers, p_val_adj < 0.05)
write.xlsx(agg.peak.markers,"global_ATAC_markers.xlsx")
agg.peak.markersv2 <- agg.peak.markers[!duplicated(agg.peak.markers$gene),]


agg.peak.markersv2 %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > .3) %>%
  slice_head(n = 50) %>%
  ungroup() -> top50

#Visualize Signature ATAC profile for each cell type (there was not one for Epicardial cells)
DoHeatmap(agg_random_IDS, group.colors = clust_colors,features = top50$gene, label = F, slot = "data",disp.max = 4)  + scale_y_discrete(labels = NULL) + scale_fill_gradient(low = "white", high = "black") + theme(legend.text =element_text(size=10, face = "bold"))
  ggsave("Peaks_cluster_heatmap.tiff", height = 5,width = 5)
  
#Visualize ATAC peaks of marker genes for each cell type
p<-CoveragePlot(
  object = aggregated_wnn,
  region = "chr19-11265000-11267000",
  features = NULL,
  expression.assay = "SCT",
  assay = "peakunion",
  links = F,
  peaks = F, extend.upstream = 0, extend.downstream = 1000
  
)
p

p7<-
  p & scale_fill_manual(values =  clust_colors) & theme(legend.text  = element_blank(),axis.title.y=element_blank(), axis.text.y=element_blank(),axis.ticks.y=element_blank(),strip.text.y.left = element_blank(),strip.background = element_blank(),axis.text.x.top  = element_text(size=8)) 

##Repeat for the following:
#p1-Myh6: chr14-54966000-54967500;p2-Pdgfra: chr5-75151500-75153000;p3-Cdh5: chr8-104101000-104103000;p4-Pdgfrb: chr18-61044500-61046500;p5- Cd14 (small gene_) chr18-36725000-36727000;p6-Upk3b: chr5-136038000-136039000;p7 -Ms4a1: chr19-11265000-11267000

p1+p2+p3+p4+p5+p6+p7

#Merge into a larger figure
combined_p <- ggarrange(p1,p2,p3,p4,p5,p6,p7, ncol = 7)
ggsave("global_ATAC.tiff", combined_p, width = 20, height = 5)


```