---
title: "SDH inhibition snMultiome RNA & ATAC-seq"
author: "Dakota Nuttall"
date: "2025-06-17"
R-version: 4.3.3
---
###Individual Sample Preparation###

```{r echo=TRUE}
set.seed(1234)
library(Seurat)
library(Signac)
library(dplyr)
library(reticulate)
library(hdf5r)
library(scales)
library(ggplot2)
library(openxlsx)
library(readxl)
library(GenomeInfoDb)
library(EnsDb.Mmusculus.v79)
library(BSgenome.Mmusculus.UCSC.mm10)
library(org.Mm.eg.db)
library(JASPAR2024)
library(TFBSTools)
library(extremevalues)
library(DropletUtils)
```
####  Load data  ####

```{r echo=TRUE}
annotation <- GetGRangesFromEnsDb(ensdb = EnsDb.Mmusculus.v79)
annotation <- renameSeqlevels(annotation, mapSeqlevels(seqlevels(annotation), "UCSC"))

####Run this in Command Prompt after cellbender to prepare for use as Seurat object:
#ptrepack --complevel 5 .\sample_CBfiltered_feature_bc_matrix.h5:/matrix sample_CBfiltered_seurat.h5:/matrix

SalSham <- "SalSham"
MalSham <- "MalSham"
SalMI <- "SalMI"
MalMI <- "MalMI"

sample_names <- c(SalSham,MalSham,SalMI,MalMI)
sample_list <- list()

  for (sample in sample_list){
  srt.dat<- Read10X_h5(paste(sample,"_CBfiltered_seurat.h5"))
srt.frag <- print(sample,"_atac_fragments.tsv.gz")
srt <- CreateSeuratObject(counts = srt.dat$`Gene Expression`,assay = "RNA",project = "Adult_snMultiome")
srt[["ATAC"]] <- CreateChromatinAssay(counts = srt.dat$Peaks,sep = c(":", "-"),fragments = srt.frag,annotation = annotation)
DefaultAssay(srt) <- "ATAC"
srt <- NucleosomeSignal(srt)
srt <- TSSEnrichment(srt)
srt <- Preprocess(srt,assay = "RNA") #See helper functions
srt <- subset(
  x = srt,
  subset = nFeature_RNA > 10 
)

srt <- RenameCells(srt, add.cell.id = sample)

sample_list[sample] <- srt

}


```

###  Set hard and dynamic quality filters

```{r echo=TRUE}
hard_cells_toss_count <- c()
hard_cells_toss_percent <- c()
dynamic_cells_toss_count <- c()
cutoff1 <- 200
cutoff2 <- 150
cutoff3 <- 5
cutoff4 <- 50
cutoff5 <- 50
n_filtered <- NULL
pct_filtered <- NULL
cells_toss_1 <-NULL
cells_toss_2 <- NULL
cells_in_group <- NULL

i <- 1


for (sample in sample_list){
  
##Hard filtering  
hard_filter <- function(sample_list)
## Filter cells with less than 200 RNA umi(counts)/cell

n_filtered <- sum(sample$nCount_RNA < cutoff1)
pct_filtered <- n_filtered*100/length(sample$nCount_RNA)

## Filter cells with less than 150 RNA genes/cell

n_filtered <- sum(sample$nFeature_RNA < cutoff2)
pct_filtered <- n_filtered*100/length(sample$nFeature_RNA)

## Filter cells with higher than 5% mitochondrial counts/cell

n_filtered <- sum(sample$percent.mt < cutoff3)
pct_filtered <- n_filtered*100/length(sample$percent.mt)

## Filter cells with less than ATAC umi(counts)/cell

n_filtered <- sum(sample$nCount_ATAC < cutoff4)
pct_filtered <- n_filtered*100/length(sample$nCount_ATAC)

## Filter cells with less than ATAC genes/cell

n_filtered <- sum(sample$nFeature_ATAC < cutoff5)
pct_filtered <- n_filtered*100/length(sample$nFeature_ATAC)

cells_toss_1 <- Cells(sample)[
        sample$nCount_RNA < cutoff1 |
                sample$nFeature_RNA < cutoff2 |
                sample$percent.mt > cutoff3  |
          sample$nCount_ATAC < cutoff4  |
          sample$nFeature_ATAC < cutoff5
]
hard_cells_toss_count <- c(hard_cells_toss_count,length(cells_toss_1)) 
hard_cells_toss_percent <- c(hard_cells_toss_percent,length(cells_toss_1)*100/length(sample$nCount_RNA)) 

sample <- sample[, !Cells(sample) %in% cells_toss_1]

## Dynamic filtering of lower 2.5% and upper 97.5%
cells_in_group <- Cells(sample)
cells_toss_2 <- c()
cutoff_upper <-  quantile(sample$nCount_RNA[cells_in_group], .975)
cutoff_lower <-  quantile(sample$nCount_RNA[cells_in_group], .025)
cells_toss_2  <- c(cells_toss_2,
                  cells_in_group[sample$nCount_RNA[cells_in_group] > cutoff_upper |
                                   sample$nCount_RNA[cells_in_group] < cutoff_lower])
dynamic_cells_toss_count <- c(dynamic_cells_toss_count,length(cells_toss_2))
 
sample_cond <- sample_names[i]
sample_list[sample_cond] <- sample[, !Cells(sample) %in% cells_toss_2]
i <- i + 1
}

```
###Use callpeaks with MACS2 to identify peaks *requires Macs/Linux*

```{r echo=TRUE}


Condition_labels <- c("Saline_Sham","Malonate_Sham","Saline_MI","Malonate_MI")
i <- 1
for (sample in sample_list){
srt <-sample
DefaultAssay(srt) <- "ATAC"
  
srt_peaks <- CallPeaks(sample, macs2.path = "~/.local/bin/macs2", outdir = "~/R/x86_64-pc-linux-gnu-library/4.3/cli/shiny/output/sample_Peak")
srt_peaks <- keepStandardChromosomes(srt_peaks, pruning.mode = "coarse")
macs_counts <- FeatureMatrix(fragments = Fragments(srt), features = srt_peaks, cells = colnames(srt))
srt[["peaks"]] <- CreateChromatinAssay(counts = macs_counts, fragments = srt.frag, annotation = annotation)
srt$orig.ident = rep(Condition_labels[i],length(srt$orig.ident))
if (i == 4){
srt$Regenerative_state = rep("Regenerative", length(srt$orig.ident))
}
else{
  srt$Regenerative_state = rep("Non-regenerative", length(srt$orig.ident))
}
  
sample_cond <- sample_names[i]
sample_list[sample_cond] <- srt

}

```

###Create a common peaks assay to account for lack of overlap between samples

```{r echo=TRUE}

peaks <- c()
for (sample in sample_list){
srt <-sample
DefaultAssay(srt) <- "peaks"

peaks <- c(peaks,StringToGRanges(regions = rownames(srt)))
}

# Create a unified set of peaks to quantify in each dataset
plan("multisession", workers = 12)
combined.peaks <- reduce(x = peaks)

# Filter out bad peaks based on length
peakwidths <- width(combined.peaks)
combined.peaks <- combined.peaks[peakwidths  < 10000 & peakwidths > 20]
combined.peaks

i <- 1
for (sample in sample_list){
  srt <-sample
# Newcounts based on unions
newcounts <- FeatureMatrix(fragments = Fragments(srt), features = combined.peaks, cells = colnames(srt))


srt[['peakunion']] <- CreateChromatinAssay(counts = newcounts1,fragments = Fragments(srt),annotation = annotation)

DefaultAssay(srt) <- "peakunion"

srt <- RunTFIDF(srt)
srt <- FindTopFeatures(srt, min.cutoff = 10)


#optiional to reduce file size
srt[['peaks']] <- NULL

sample_cond <- sample_names[i]
sample_list[sample_cond] <- srt

}

```

##Normalize RNA counts with SCTransform and integrate datasets

```{r echo=TRUE}
##Normalize 
i <- 1
for (sample in sample_list){
  srt <-sample
salSham <-  SCTransform(salSham,vars.to.regress = "percent.mt",vst.flavor = "v2", method = "glmGamPoi",variable.features.n = 5000)

DefaultAssay(srt) <- "SCT"

sample_cond <- sample_names[i]
sample_list[sample_cond] <- srt

}

##Merge using integration anchors 
options(future.globals.maxSize = 40000 * 1024^2)

features <- SelectIntegrationFeatures(object.list = sample_list, nfeatures = 5000)
agg.list <- PrepSCTIntegration(object.list = agg.list, anchor.features = features)
agg.anchors <- FindIntegrationAnchors(object.list = agg.list, normalization.method = "SCT", anchor.features = features)
aggregated_samples <- IntegrateData(anchorset = agg.anchors, normalization.method = "SCT")

#Save integrated data
saveRDS(aggregated_samples, file = "aggregated_Integrated.rds")



```
