---
title: "SDH inhibition snMultiome RNA & ATAC-seq"
author: "Dakota Nuttall"
date: "2025-06-17"
R-version: 4.3.3
---
###Myeloid, Endothelial Cell, Mural and Lymph Analysis###

```{r echo=TRUE}
set.seed(1234)
library(Seurat)
library(Signac)
library(chromVAR)
library(dplyr)
library(reticulate)
library(hdf5r)
library(scales)
library(ggplot2)
library(openxlsx)
library(readxl)
library(EnsDb.Mmusculus.v79)
library(BSgenome.Mmusculus.UCSC.mm10)
library(org.Mm.eg.db)
library(JASPAR2020)
library(TFBSTools)
library(harmony)
library(pheatmap)
library(EnhancedVolcano)
library(ggrepel)
library(tidyr)
library(gridExtra)
library(ggpubr)
```
####  Load data  ####

```{r echo=TRUE}
#Load integrated file
aggregated_samples <- readRDS("aggregated_wnn.rds")

clust_colors <- c("darkorange2","mediumorchid2","steelblue3","olivedrab","indianred1","hotpink","darkred","green3","grey")
treatment_colors <-c("lightskyblue","pink", "steelblue4","red3")    
```
## Myeloid Cells
```{r echo=TRUE}

#Subset Myeloid (Mye) population

aggregated_Myeloid <- subset(aggregated_samples, idents = "Myeloid Cells")
aggregated_Myeloid <- SCTransform(aggregated_Myeloid, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)

DefaultAssay(aggregated_Myeloid) <- "SCT"

FindDimNumber(aggregated_Myeloid)

ElbowPlot(aggregated_Myeloid, ndims = 50, reduction = "lsi")
aggregated_Myeloid <- RunUMAP(aggregated_Myeloid, dims = 1:50, reduction = "harmony_pca",reduction.name = "umap_RNA", reduction.key = "UMAP")

DefaultAssay(aggregated_Myeloid) <- "peakunion"
aggregated_Myeloid <- FindTopFeatures(aggregated_Myeloid, min.cutoff = 10)
aggregated_Myeloid<- RunTFIDF(aggregated_Myeloid)
aggregated_Myeloid <- RunSVD(aggregated_Myeloid)

ElbowPlot(aggregated_Myeloid, ndims = 50, reduction = "lsi")

aggregated_Myeloid <- RunUMAP(object = aggregated_Myeloid, reduction = 'harmony_lsi', dims = 2:50, reduction.name = "umap_peakunion", reduction.key = "peakunionUMAP_")

DefaultAssay(aggregated_Myeloid) <- "SCT"
aggregated_Myeloid <- FindMultiModalNeighbors(
  object = aggregated_Myeloid,
  reduction.list = list("harmony_pca", "harmony_lsi"), 
  dims.list = list(1:40, 2:20),# Determinded to be optimal amount of dims through visualization
  modality.weight.name = c("SCT.weight","peakunion.weight"),
  verbose = TRUE
)
aggregated_Myeloid <- RunUMAP(
  object = aggregated_Myeloid,
  nn.name = "weighted.nn",
  assay = "SCT",
  reduction.name = "WNN_umap",
  reduction.key = "WNN_UMAP_",
  verbose = TRUE,
)

aggregated_Myeloid<- FindClusters(aggregated_Myeloid, graph.name = "wsnn", algorithm = 3, resolution = seq(.1,.5,.1) )

quick_CT_plot(aggregated_Myeloid, 2,"WNN_umap",legend = F,label = T,font.size = 8,resolution = "SCT_snn_res.0.5")
quick_cond_plot(aggregated_Myeloid, 2,"WNN_umap",legend = F)

#Reference dataframe obtained from our collaborators from the Xiao Li lab
reference_db <- read.csv("PART90.02.05.table.Immune_cell_state_RNA_marker.csv")

aggregated_Myeloid <- PrepSCTFindMarkers(aggregated_Myeloid, assay = "SCT",verbose = T)
Myeloid.markers <- FindAllMarkers(aggregated_Myeloid, only.pos = TRUE, logfc.threshold = .3, return.thresh = 0.1, min.pct = 0.1, test.use = "MAST", recorrect_umi = TRUE)

Myeloid.markers <- subset(Myeloid.markers, p_val_adj < 0.05)

assigned_cell_types <- data.frame()

for (cluster_id in unique(Myeloid.markers$cluster)) {
  cluster_markers <- subset(Myeloid.markers, cluster == cluster_id)$gene
  cluster_cell_types <- reference_db$Cell_type[match(cluster_markers, reference_db$Gene_symbol, nomatch = NA)]
  cluster_assignment <- data.frame(Cluster = cluster_id, CellType = cluster_cell_types, Gene = cluster_markers)
  assigned_cell_types <- rbind(assigned_cell_types, cluster_assignment)
}

Myeloid.anno <- cbind(Myeloid.markers,assigned_cell_types$CellType)

Myeloid.anno_filtered <- na.omit(Myeloid.anno)
clustergenes2 <-Myeloid.anno_filtered %>%
  group_by(cluster) %>%
  slice_min(n = 20, order_by =p_val_adj)

write.csv(Myeloid.anno,file = "myeloid.anno.csv")

Myeloid_cell_types <- c("MP1","MP3","Mono","MP2","MP4")
aggregated_Myeloid@active.ident<-factor(aggregated_Myeloid@active.ident, labels = Myeloid_cell_types)

Myeloid_cell_levels <- c("MP1","MP2","MP3","MP4","Mono")
aggregated_Myeloid@active.ident<-factor(aggregated_Myeloid@active.ident, levels = Myeloid_cell_levels)

aggregated_Myeloid$Cell_state <- Idents(aggregated_Myeloid)
#These functions can be found in the helper functions script
proportion_plots(aggregated_Myeloid, cond_color = treatment_colors, clust_color = clust_colors, name.clust = "Mye_prop_sub.tiff", name.cond = "Mye_prop_cond.tiff",height = 4, width = 4)

prop_heatmap(aggregated_Myeloid, "Mye_heat_prop.tiff")

quick_CT_plot(aggregated_Myeloid, 2,"WNN_umap",legend = F,label = T, font.size = 8)
ggsave("Myeloid_Cells_umap_sub.tiff", height = 5,width = 5)
quick_cond_plot(aggregated_Myeloid_Cells, 2,"WNN_umap",legend = F)
ggsave("Myeloid_Cells_umap_cond.tiff", height = 5,width = 5)

#Save dataset
save(aggregated_Myeloid,"Myeloid.rds")

```
## Endothelial Cells
```{r echo=TRUE}

#Subset Endothelial_Cells (EC) population

aggregated_Endothelial_Cells <- subset(aggregated_samples, idents = "Endothelial Cells")
aggregated_Endothelial_Cells <- SCTransform(aggregated_Endothelial_Cells, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)

DefaultAssay(aggregated_Endothelial_Cells) <- "SCT"

FindDimNumber(aggregated_Endothelial_Cells)

ElbowPlot(aggregated_Endothelial_Cells, ndims = 50, reduction = "lsi")
aggregated_Endothelial_Cells <- RunUMAP(aggregated_Endothelial_Cells, dims = 1:50, reduction = "harmony_pca",reduction.name = "umap_RNA", reduction.key = "UMAP")

DefaultAssay(aggregated_Endothelial_Cells) <- "peakunion"
aggregated_Endothelial_Cells <- FindTopFeatures(aggregated_Endothelial_Cells, min.cutoff = 10)
aggregated_Endothelial_Cells<- RunTFIDF(aggregated_Endothelial_Cells)
aggregated_Endothelial_Cells <- RunSVD(aggregated_Endothelial_Cells)

ElbowPlot(aggregated_Endothelial_Cells, ndims = 50, reduction = "lsi")

aggregated_Endothelial_Cells <- RunUMAP(object = aggregated_Endothelial_Cells, reduction = 'harmony_lsi', dims = 2:50, reduction.name = "umap_peakunion", reduction.key = "peakunionUMAP_")

DefaultAssay(aggregated_Endothelial_Cells) <- "SCT"
aggregated_Endothelial_Cells <- FindMultiModalNeighbors(
  object = aggregated_Endothelial_Cells,
  reduction.list = list("harmony_pca", "harmony_lsi"), 
  dims.list = list(1:50, 2:20),# Determinded to be optimal amount of dims through visualization
  modality.weight.name = c("SCT.weight","peakunion.weight"),
  verbose = TRUE
)
aggregated_Endothelial_Cells <- RunUMAP(
  object = aggregated_Endothelial_Cells,
  nn.name = "weighted.nn",
  assay = "SCT",
  reduction.name = "WNN_umap",
  reduction.key = "WNN_UMAP_",
  verbose = TRUE,
)

aggregated_Endothelial_Cells<- FindClusters(aggregated_Endothelial_Cells, graph.name = "wsnn", algorithm = 3, resolution = seq(.1,.5,.1) )

quick_CT_plot(aggregated_Endothelial_Cells, 2,"WNN_umap",legend = F,label = T,font.size = 8,resolution = "SCT_snn_res.0.3")
quick_cond_plot(aggregated_Endothelial_Cells, 2,"WNN_umap",legend = F)

#Reference dataframe obtained from our collaborators from the Xiao Li lab
reference_db <- read.csv("PART90.02.03.table.ec_cell_state_RNA_marker.csv")

aggregated_Endothelial_Cells <- PrepSCTFindMarkers(aggregated_Endothelial_Cells, assay = "SCT",verbose = T)
Endothelial_Cells.markers <- FindAllMarkers(aggregated_Endothelial_Cells, only.pos = TRUE, logfc.threshold = .3, return.thresh = 0.1, min.pct = 0.1, test.use = "MAST", recorrect_umi = TRUE)

Endothelial_Cells.markers <- subset(Endothelial_Cells.markers, p_val_adj < 0.05)

assigned_cell_types <- data.frame()

for (cluster_id in unique(Endothelial_Cells.markers$cluster)) {
  cluster_markers <- subset(Endothelial_Cells.markers, cluster == cluster_id)$gene
  cluster_cell_types <- reference_db$Cell_type[match(cluster_markers, reference_db$Gene_symbol, nomatch = NA)]
  cluster_assignment <- data.frame(Cluster = cluster_id, CellType = cluster_cell_types, Gene = cluster_markers)
  assigned_cell_types <- rbind(assigned_cell_types, cluster_assignment)
}

Endothelial_Cells.anno <- cbind(Endothelial_Cells.markers,assigned_cell_types$CellType)

Endothelial_Cells.anno_filtered <- na.omit(Endothelial_Cells.anno)
clustergenes2 <-Endothelial_Cells.anno_filtered %>%
  group_by(cluster) %>%
  slice_min(n = 20, order_by =p_val_adj)

write.csv(Endothelial_Cells.anno,file = "Endothelial_Cells.anno.csv")

Endothelial_Cells_cell_types <- c("VEC3","VEC2","VEC_Prol1","VEC4","EndoC","VEC1","LEC","VEC_Prol2")
aggregated_Endothelial_Cells@active.ident<-factor(aggregated_Endothelial_Cells@active.ident, labels = Endothelial_Cells_cell_types)
Endothelial_Cells_cell_levels <- c("VEC1","VEC2","VEC3","VEC4","VEC_Prol1","VEC_Prol2","EndoC","LEC")
aggregated_Endothelial_Cells@active.ident<-factor(aggregated_Endothelial_Cells@active.ident, levels = Endothelial_Cells_cell_levels)

aggregated_Endothelial_Cells$Cell_state <- Idents(aggregated_Endothelial_Cells)
#These functions can be found in the helper functions script
proportion_plots(aggregated_Endothelial_Cells, cond_color = treatment_colors, clust_color = clust_colors, name.clust = "EC_prop_sub.tiff", name.cond = "EC_prop_cond.tiff",height = 4, width = 4)

prop_heatmap(aggregated_Endothelial_Cells, "EC_heat_prop.tiff")

quick_CT_plot(aggregated_Endothelial_Cells, 2,"WNN_umap",legend = F,label = T, font.size = 8)
ggsave("Endothelial_Cells_umap_sub.tiff", height = 5,width = 5)
quick_cond_plot(aggregated_Endothelial_Cells, 2,"WNN_umap",legend = F)
ggsave("Endothelial_Cells_umap_cond.tiff", height = 5,width = 5)

#Save dataset
save(aggregated_Endothelial_Cells,"Endothelial_Cells.rds")

```
## Mural Cells
```{r echo=TRUE}

#Subset Mural (Mural) population

aggregated_Mural <- subset(aggregated_samples, idents = "Mural Cells")
aggregated_Mural <- SCTransform(aggregated_Mural, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)

DefaultAssay(aggregated_Mural) <- "SCT"

FindDimNumber(aggregated_Mural)

ElbowPlot(aggregated_Mural, ndims = 50, reduction = "lsi")
aggregated_Mural <- RunUMAP(aggregated_Mural, dims = 1:50, reduction = "harmony_pca",reduction.name = "umap_RNA", reduction.key = "UMAP")

DefaultAssay(aggregated_Mural) <- "peakunion"
aggregated_Mural <- FindTopFeatures(aggregated_Mural, min.cutoff = 10)
aggregated_Mural<- RunTFIDF(aggregated_Mural)
aggregated_Mural <- RunSVD(aggregated_Mural)

ElbowPlot(aggregated_Mural, ndims = 50, reduction = "lsi")

aggregated_Mural <- RunUMAP(objMuralt = aggregated_Mural, reduction = 'harmony_lsi', dims = 2:50, reduction.name = "umap_peakunion", reduction.key = "peakunionUMAP_")

DefaultAssay(aggregated_Mural) <- "SCT"
aggregated_Mural <- FindMultiModalNeighbors(
  objMuralt = aggregated_Mural,
  reduction.list = list("harmony_pca", "harmony_lsi"), 
  dims.list = list(1:50, 2:10),# Determinded to be optimal amount of dims through visualization
  modality.weight.name = c("SCT.weight","peakunion.weight"),
  verbose = TRUE
)
aggregated_Mural <- RunUMAP(
  objMuralt = aggregated_Mural,
  nn.name = "weighted.nn",
  assay = "SCT",
  reduction.name = "WNN_umap",
  reduction.key = "WNN_UMAP_",
  verbose = TRUE,
)

aggregated_Mural<- FindClusters(aggregated_Mural, graph.name = "wsnn", algorithm = 3, resolution = seq(.1,.5,.1) )

quick_CT_plot(aggregated_Mural, 2,"WNN_umap",legend = F,label = T,font.size = 8,resolution = "SCT_snn_res.0.4")
quick_cond_plot(aggregated_Mural, 2,"WNN_umap",legend = F)

#Reference dataframe obtained from our collaborators from the Xiao Li lab
reference_db <- read.csv("PART90.02.04.table.mural_cell_state_RNA_marker.csv")

aggregated_Mural <- PrepSCTFindMarkers(aggregated_Mural, assay = "SCT",verbose = T)
Mural.markers <- FindAllMarkers(aggregated_Mural, only.pos = TRUE, logfc.threshold = .3, return.thresh = 0.1, min.pct = 0.1, test.use = "MAST", rMuralorrMuralt_umi = TRUE)

Mural.markers <- subset(Mural.markers, p_val_adj < 0.05)

assigned_cell_types <- data.frame()

for (cluster_id in unique(Mural.markers$cluster)) {
  cluster_markers <- subset(Mural.markers, cluster == cluster_id)$gene
  cluster_cell_types <- reference_db$Cell_type[match(cluster_markers, reference_db$Gene_symbol, nomatch = NA)]
  cluster_assignment <- data.frame(Cluster = cluster_id, CellType = cluster_cell_types, Gene = cluster_markers)
  assigned_cell_types <- rbind(assigned_cell_types, cluster_assignment)
}

Mural.anno <- cbind(Mural.markers,assigned_cell_types$CellType)

Mural.anno_filtered <- na.omit(Mural.anno)
clustergenes2 <-Mural.anno_filtered %>%
  group_by(cluster) %>%
  slice_min(n = 20, order_by =p_val_adj)

write.csv(Mural.anno,file = "Mural.anno.csv")

Mural_cell_types <- c("PC1","PC1","PC1","PC1","PC2","SMC")
aggregated_Mural@active.ident<-factor(aggregated_Mural@active.ident, labels = Mural_cell_types)


aggregated_Mural$Cell_state <- Idents(aggregated_Mural)
#These functions can be found in the helper functions script
proportion_plots(aggregated_Mural, cond_color = treatment_colors, clust_color = clust_colors, name.clust = "Mural_prop_sub.tiff", name.cond = "Mural_prop_cond.tiff",height = 4, width = 4)

prop_heatmap(aggregated_Mural, "Mural_heat_prop.tiff")

quick_CT_plot(aggregated_Mural, 2,"WNN_umap",legend = F,label = T, font.size = 8)
ggsave("Mural_Cells_umap_sub.tiff", height = 5,width = 5)
quick_cond_plot(aggregated_Mural, 2,"WNN_umap",legend = F)
ggsave("Mural_Cells_umap_cond.tiff", height = 5,width = 5)

#Save dataset
save(aggregated_Mural,"Mural.rds")

```
## Lymphoid Cells
```{r echo=TRUE}

#Subset Lymphoid (Lymph) population

aggregated_Lymphoid <- subset(aggregated_samples, idents = "Lymphoid Cells")
aggregated_Lymphoid <- SCTransform(aggregated_Lymphoid, vars.to.regress = c("percent.mt","percent.ribo"),vst.flavor = "v2", method = "glmGamPoi", variable.features.n = 5000)

DefaultAssay(aggregated_Lymphoid) <- "SCT"

FindDimNumber(aggregated_Lymphoid)

ElbowPlot(aggregated_Lymphoid, ndims = 50, reduction = "lsi")
aggregated_Lymphoid <- RunUMAP(aggregated_Lymphoid, dims = 1:50, reduction = "harmony_pca",reduction.name = "umap_RNA", reduction.key = "UMAP")

DefaultAssay(aggregated_Lymphoid) <- "peakunion"
aggregated_Lymphoid <- FindTopFeatures(aggregated_Lymphoid, min.cutoff = 10)
aggregated_Lymphoid<- RunTFIDF(aggregated_Lymphoid)
aggregated_Lymphoid <- RunSVD(aggregated_Lymphoid)

ElbowPlot(aggregated_Lymphoid, ndims = 50, reduction = "lsi")

aggregated_Lymphoid <- RunUMAP(objLymphoidt = aggregated_Lymphoid, reduction = 'harmony_lsi', dims = 2:50, reduction.name = "umap_peakunion", reduction.key = "peakunionUMAP_")

DefaultAssay(aggregated_Lymphoid) <- "SCT"
aggregated_Lymphoid <- FindMultiModalNeighbors(
  objLymphoidt = aggregated_Lymphoid,
  reduction.list = list("harmony_pca", "harmony_lsi"), 
  dims.list = list(1:50, 2:7),# Determinded to be optimal amount of dims through visualization
  modality.weight.name = c("SCT.weight","peakunion.weight"),
  verbose = TRUE
)
aggregated_Lymphoid <- RunUMAP(
  objLymphoidt = aggregated_Lymphoid,
  nn.name = "weighted.nn",
  assay = "SCT",
  reduction.name = "WNN_umap",
  reduction.key = "WNN_UMAP_",
  verbose = TRUE,
)

aggregated_Lymphoid<- FindClusters(aggregated_Lymphoid, graph.name = "wsnn", algorithm = 3, resolution = seq(.1,.5,.1) )

quick_CT_plot(aggregated_Lymphoid, 2,"WNN_umap",legend = F,label = T,font.size = 8,resolution = "SCT_snn_res.0.4")
quick_cond_plot(aggregated_Lymphoid, 2,"WNN_umap",legend = F)

#Reference dataframe obtained from our collaborators from the Xiao Li lab
reference_db <- reference_db <- read.csv("PART90.02.05.table.Immune_cell_state_RNA_marker.csv")

aggregated_Lymphoid <- PrepSCTFindMarkers(aggregated_Lymphoid, assay = "SCT",verbose = T)
Lymphoid.markers <- FindAllMarkers(aggregated_Lymphoid, only.pos = TRUE, logfc.threshold = .3, return.thresh = 0.1, min.pct = 0.1, test.use = "MAST", rLymphoidorrLymphoidt_umi = TRUE)

Lymphoid.markers <- subset(Lymphoid.markers, p_val_adj < 0.05)

assigned_cell_types <- data.frame()

for (cluster_id in unique(Lymphoid.markers$cluster)) {
  cluster_markers <- subset(Lymphoid.markers, cluster == cluster_id)$gene
  cluster_cell_types <- reference_db$Cell_type[match(cluster_markers, reference_db$Gene_symbol, nomatch = NA)]
  cluster_assignment <- data.frame(Cluster = cluster_id, CellType = cluster_cell_types, Gene = cluster_markers)
  assigned_cell_types <- rbind(assigned_cell_types, cluster_assignment)
}

Lymphoid.anno <- cbind(Lymphoid.markers,assigned_cell_types$CellType)

Lymphoid.anno_filtered <- na.omit(Lymphoid.anno)
clustergenes2 <-Lymphoid.anno_filtered %>%
  group_by(cluster) %>%
  slice_min(n = 20, order_by =p_val_adj)

write.csv(Lymphoid.anno,file = "Lymphoid.anno.csv")

Lymph_cell_types <- c("TC","BC")
aggregated_Lymphoid@active.ident<-factor(aggregated_Lymphoid@active.ident, labels = Lymph_cell_types)

aggregated_Lymphoid$Cell_state <- Idents(aggregated_Lymphoid)

quick_CT_plot(aggregated_Lymphoid, 2,"WNN_umap",legend = F,label = T, font.size = 8)
ggsave("Lymphoid_Cells_umap_sub.tiff", height = 5,width = 5)
quick_cond_plot(aggregated_Lymphoid, 2,"WNN_umap",legend = F)
ggsave("Lymphoid_Cells_umap_cond.tiff", height = 5,width = 5)

#Save dataset
save(aggregated_Lymphoid,"Lymphoid.rds")

```
